### PR TITLE
Delete temp tables

### PR DESCRIPTION
**Pull request by tpetracca**
Blocked on anything flagged in #915 (and this addresses that issue)
CC @ashrayjain @clockfort

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/916)

### CODE REVIEW

**Comment by clockfort** [comment]
+1 lgtm

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java [hunk 1]
```diff
@@ -206,7 +206,7 @@ protected SnapshotTransaction createTransaction(long immutableLockTs,
                 cleaner.getTransactionReadTimeoutMillis(),
                 TransactionReadSentinelBehavior.THROW_EXCEPTION,
                 allowHiddenTableAccess);
-        return runTaskThrowOnConflict(task, new OnlyWriteTempTablesTransaction(transaction, sweepStrategyManager));
+        return runTaskThrowOnConflict(task, transaction);
```
Comment (clockfort) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java]: I think this is the line causing your test failures, I suspect `transaction` here needs to be wrapped to make this a read only transaction

**Comment by tpetracca** [comment]
Ok so ReadOnlyTransaction in no way actually enforces that it's read only . . sort of scary.

**Comment by clockfort** [comment]
Yep. That's the next change to make. You wanna do it or me?

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/ReadOnlyTransaction.java [hunk 2]
```diff
@@ -1,141 +0,0 @@
...
-    }
-
-    @Override
-    protected boolean shouldDeleteAndRollback() {
-        // We don't want to delete any data or roll back any transactions because we don't participate in the
-        // transaction protocol.  We just want to skip over anything we find that isn't committed
-        return false;
-    }
-
-    public BatchingVisitable<RowResult<Value>> getRangeWithTimestamps(TableReference tableRef, RangeRequest range) {
```
Comment (clockfort) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/ReadOnlyTransaction.java]: Interesting. Looks like even in pre-forked Atlas this method wasn't used.

**Comment by clockfort** [comment]
lgtm

**Comment by clockfort** [comment]
Reviewed 2 of 4 files at r1, 1 of 1 files at r2, 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/916)*

**Pull request merged**